### PR TITLE
improve url validation in settings

### PR DESCRIPTION
**Pull request by jcalado**
It doesn't look to me like URI.tryParse is doing enough to verify if a URL is ok.
It might be better to just check against a regex like `https?:\/\/`

### CODE REVIEW

**Review by austinried — CHANGES_REQUESTED** [review]
The problem with bypassing the `Uri.tryParse()` check is going to be further down the line when `Uri.parse()` is used, since the HttpClient relies on a valid `Uri` for requests.

Can you give some specific examples of URLs that don't pass here but should work?

**Comment by jcalado** [comment]
Everything that does not specifically mention protocol. If you only set it
to host.tld it doesn't work. http://host.tld does, and so does
https://host.tld

**Comment by austinried** [comment]
Ah yeah, I see now that's actually passing the `Uri.tryParse()` but not working later with the HTTP client in those cases. In 1.x this is solved by adding the scheme to the URL if it's missing, I think we could do something similar here by try/catching `Uri.parse()` and checking the scheme, then prepending a default if it's missing.

**Comment by austinried** [comment]
Actually thinking about this more, even without automatically adding the scheme this does seem like an improvement on the existing validation so we can merge it in.

File: lib/app/pages/source_page.dart [hunk 1]
```diff
@@ -44,10 +44,10 @@ class SourcePage extends HookConsumerWidget {
       autofillHints: const [AutofillHints.url],
       required: true,
       validator: (value, label) {
-        if (Uri.tryParse(value!) == null) {
+        if (Uri.tryParse(value!)?.isAbsolute == false ||
+            !value.contains('http')) {
           return '$label must be a valid URL';
         }
-        return null;
```
Comment (austinried) [inline lib/app/pages/source_page.dart:50]: Looks like we need to keep this line though for the analyzer.

**Review by austinried — CHANGES_REQUESTED** [review]

File: lib/app/pages/source_page.dart (hunk 1 above)
Comment (jcalado) [reply lib/app/pages/source_page.dart:50]: @austinried should be fixed now

**Review by austinried — APPROVED** [review]

**Comment by austinried** [comment]
Looks good, thank you!

**Pull request merged**